app/controllers/saml_idp_controller.rb

```diff
@@ -2,6 +2,8 @@
 require 'saml_idp'
 
 class SamlIdpController < ApplicationController
+  before_action :set_devise_failure_redirect_for_concurrent_session_logout, only: [:auth, :logout]
```

Comment:
Ordering is important since the concurrent session logout happens as soon as there's any reference to the user (via [`after_set_user` hook](https://github.com/18F/identity-idp/blob/6abbb42ba14f0af05dde56f819d3ee17e3714a00/config/initializers/session_limitable.rb#L16)). Since this is now applied to the `:auth` action, it had to be moved up here since [the `include`'d `SamlIdpAuthConcern` makes references to the user for the `:auth` action](https://github.com/18F/identity-idp/blob/main/app/controllers/concerns/saml_idp_auth_concern.rb).

Reply:
if order matters here should we add a comment describing that?

Reply:
Sure, I'll add one. I think I subconsciously assumed the feature tests would suffice to catch any regressions if someone were to shift it around, but it could be helpful to have an inline comment explaining it, especially if I felt compelled to leave a GitHub comment doing so 😅

Reply:
Updated in 9ab88cb41